From a57d79732c60705243b93f8cb5cd5695dc43c05a Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Fri, 17 Nov 2017 16:41:32 +0800
Subject: [PATCH 2833/5242] MLK-16839-3: ASoC: fsl_esai: remove rate constrain
 for FE-BE case

commit  9a3195ab4ee4822fd83f0e93d6b17ec7ac585dda from
https://source.codeaurora.org/external/imx/linux-imx.git

In FE-BE case, the constraint in BE cpu dai will propagate to FE.
and the rate constaint is set by .be_hw_params_fixup in dai_link for BE.
So rate constraint in BE cpu dai is not needed in FE-BE case.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_esai.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/sound/soc/fsl/fsl_esai.c b/sound/soc/fsl/fsl_esai.c
index 6370d59..cd0020e 100644
--- a/sound/soc/fsl/fsl_esai.c
+++ b/sound/soc/fsl/fsl_esai.c
@@ -538,6 +538,7 @@ static int fsl_esai_startup(struct snd_pcm_substream *substream,
 			    struct snd_soc_dai *dai)
 {
 	struct fsl_esai *esai_priv = snd_soc_dai_get_drvdata(dai);
+	struct snd_soc_pcm_runtime *rtd = substream->private_data;
 	bool tx = substream->stream == SNDRV_PCM_STREAM_PLAYBACK;
 	int ret;
 
@@ -593,7 +594,9 @@ static int fsl_esai_startup(struct snd_pcm_substream *substream,
 	if (esai_priv->soc->dma_workaround) {
 		snd_pcm_hw_constraint_minmax(substream->runtime,
 				SNDRV_PCM_HW_PARAM_CHANNELS, 1, 2);
-		snd_pcm_hw_constraint_minmax(substream->runtime,
+
+		if (!rtd->dai_link->be_hw_params_fixup)
+			snd_pcm_hw_constraint_minmax(substream->runtime,
 				SNDRV_PCM_HW_PARAM_RATE, 48000, 48000);
 	}
 
-- 
1.7.9.5

